tests/test_histogram.py

```diff
@@ -20,6 +20,82 @@
 )
 
 
+class TestDigitize:
+    @pytest.mark.parametrize(
+        "dtype", get_all_dtypes(no_bool=True, no_complex=True)
+    )
+    @pytest.mark.parametrize("right", [True, False])
+    @pytest.mark.parametrize(
+        "x, bins",
+        [
+            # Negative values
+            (
+                numpy.array([-5, -3, -1, 0, 1, 3, 5]),
+                numpy.array([-4, -2, 0, 2, 4]),
+            ),
+            # Non-uniform bins
+            (
+                numpy.array([1, 2, 3, 4, 5, 6, 7, 8, 9]),
+                numpy.array([1, 4, 6, 7]),
+            ),
+            # Infinity values
+            (
+                numpy.array([-numpy.inf, -1, 0, 1, numpy.inf]),
+                numpy.array([-2, -1, 0, 1, 2]),
+            ),
+            # Repeated elements
+            (numpy.array([1, 2, 2, 3, 3, 3, 4, 5]), numpy.array([1, 2, 3, 4])),
+        ],
+    )
+    def test_digitize(self, x, bins, dtype, right):
+        x = x.astype(dtype)
+        bins = bins.astype(dtype)
+        x_dp = dpnp.array(x)
+        bins_dp = dpnp.array(bins)
+
+        result = dpnp.digitize(x_dp, bins_dp, right=right)
+        expected = numpy.digitize(x, bins, right=right)
+        assert_dtype_allclose(result, expected)
+
+    @pytest.mark.parametrize(
+        "dtype", get_all_dtypes(no_bool=True, no_complex=True)
+    )
+    @pytest.mark.parametrize(
+        "x, bins",
+        [
+            # Empty array
+            (numpy.array([]), numpy.array([1, 2, 3])),
+            # Empty bins
+            (numpy.array([1, 2, 3]), numpy.array([])),
+        ],
+    )
+    def test_digitize_empty(self, x, bins, dtype):
```

Comment:
Do we need to check a use case when `x` and `bins` have different data types?

Reply:
It makes sense!